### DISCUSSION TITLE
Difference between nvim-surround and vim-surround

### DISCUSSION THREAD

**Discussion by perseoGI** [kylechui/nvim-surround#55] [Q&A] [answered]
Hi!
What are the main differences between this plugin and the old school vim-surround (apart from being written in lua)?

**Comment by kylechui** [comment, marked as answer]
Hi there! Here's a list of a few things that stand out to me (hopefully others will add to this list if I forget anything):
* Inserting function-evaluated surrounds
  * This allows for processing to be done while the surround is being created, e.g. getting user input, calling into Tree-sitter, etc.
  * Some examples of this can be found [in the showcase page](https://github.com/kylechui/nvim-surround/discussions/53)
* When changing/deleting, jumping can occur backwards if no surrounding pairs are found around the cursor nor after, so if the cursor is in the comment for the below example, then `ds"` will delete the quotes anyhow
```lua
local str = "some text" -- with a comment
```
* Character "aliasing" allows you to let one char stand in for several other characters, e.g. by default `q` is mapped to ``{ `, ', " }``. That way `dsq` will delete the *nearest* set of quotes of any kind, which ties in with dot-repeating quite nicely I think. See the README's video for an example.
* Highlighting inserts/changes can be seen in the README's demo video, which can be nice for checking yourself

**Reply by perseoGI** [reply]
This is pretty neat! I have already uninstalled old tpope's and installed yours.
Love each of the points you answer! 
Great job :rocket:

**Reply by ariel-frischer** [reply]
Is there anything this plugin does not have but tpope/vim-surround has? Also this should be faster because its written in LUA right?

**Reply by kylechui** [reply]
Hi there, so far I think I've covered most (if not all) of what tpope's vim-surround has (at the very least I haven't gotten a feature parity request in a while). I will say that speed shouldn't really be a deciding factor, as the actual computations performed in both aren't very intensive. If you're curious, I would recommend trying out nvim-surround as a drop-in replacement and see how it works for yourself. The main two benefits would be more functionality (e.g. `dsf` for deleting function calls), as well as a saner configuration language. If you have any more questions, feel free to ask!
